ui/app/(ai)/analyst/route.ts

```diff
@@ -0,0 +1,267 @@
+    ? `- Latest Scan ID: ${provider.scan_id}
+- Scan Duration: ${provider.scan_duration || "Unknown"}
+- Resource Count: ${provider.resource_count || "Unknown"}`
+    : "- No completed scans found"
+}
+`,
+  )
+  .join("\n")}
+`;
+    }
+    return "";
+  } catch (error) {
+    console.error("Failed to retrieve cached data:", error);
+    return "**CURRENT DATA: Not available**";
+  }
+};
+
+const initializeModels = async () => {
+  const apiKey = await getAIKey();
+  const aiConfig = await getAIConfiguration();
+  const modelConfig = aiConfig?.data?.attributes;
+
+  // Initialize models without API keys
+  const llm = new ChatOpenAI({
+    model: modelConfig?.model || "gpt-4o",
+    temperature: modelConfig?.temperature || 0,
+    maxTokens: modelConfig?.max_tokens || 4000,
+    apiKey: apiKey,
+    tags: ["agent"],
+  });
+
+  const supervisorllm = new ChatOpenAI({
+    model: modelConfig?.model || "gpt-4o",
+    temperature: modelConfig?.temperature || 0,
+    maxTokens: modelConfig?.max_tokens || 4000,
+    apiKey: apiKey,
+    streaming: true,
+    tags: ["supervisor"],
+  });
+
+  const providerAgent = createReactAgent({
+    llm: llm,
+    tools: [getProvidersTool, getProviderTool],
+    name: "provider_agent",
+    prompt: providerAgentPrompt,
+  });
+
+  const userInfoAgent = createReactAgent({
+    llm: llm,
+    tools: [getUsersTool, getMyProfileInfoTool],
+    name: "user_info_agent",
+    prompt: userInfoAgentPrompt,
+  });
+
+  const scansAgent = createReactAgent({
+    llm: llm,
+    tools: [getScansTool, getScanTool],
+    name: "scans_agent",
+    prompt: scansAgentPrompt,
+  });
+
+  const complianceAgent = createReactAgent({
+    llm: llm,
+    tools: [
+      getCompliancesOverviewTool,
+      getComplianceOverviewTool,
+      getComplianceFrameworksTool,
+    ],
+    name: "compliance_agent",
+    prompt: complianceAgentPrompt,
+  });
+
+  const findingsAgent = createReactAgent({
+    llm: llm,
+    tools: [getFindingsTool, getMetadataInfoTool, getProviderChecksTool],
+    name: "findings_agent",
+    prompt: findingsAgentPrompt,
+  });
+
+  const overviewAgent = createReactAgent({
+    llm: llm,
+    tools: [
+      getProvidersOverviewTool,
+      getFindingsByStatusTool,
+      getFindingsBySeverityTool,
+    ],
+    name: "overview_agent",
+    prompt: overviewAgentPrompt,
+  });
+
+  const rolesAgent = createReactAgent({
+    llm: llm,
+    tools: [getRolesTool, getRoleTool],
+    name: "roles_agent",
+    prompt: rolesAgentPrompt,
+  });
+
+  const agents = [
+    userInfoAgent,
+    providerAgent,
+    overviewAgent,
+    scansAgent,
+    complianceAgent,
+    findingsAgent,
+    rolesAgent,
+  ];
+
+  // Create supervisor workflow
+  const workflow = createSupervisor({
+    agents: agents,
+    llm: supervisorllm,
+    prompt: supervisorPrompt,
+    outputMode: "last_message",
+  });
+
+  // Compile and run
+  const app = workflow.compile();
+  return app;
+};
+
+export async function POST(req: Request) {
```

Comment:
This should go inside `app/api/lighthouse/yourFilesHere`

For references: https://nextjs.org/docs/14/app/api-reference

Reply:
Moved routing code to `/app/api/lighthouse/analyst/route.ts`. All other code is stored in their respective files at `/lib/lighthouse` files.